beacon-chain/blockchain/chain_info.go

```diff
@@ -189,6 +190,14 @@ func (s *Service) HeadETH1Data() *ethpb.Eth1Data {
 	return s.head.state.Eth1Data()
 }
 
+// HeadFinalizedEpoch returns the finalized epoch of the current head state.
+func (s *Service) HeadFinalizedEpoch() uint64 {
```

Comment:
I think we should be using the latest cached finalized epoch. Not the finalized epoch from the head. Keep in mind, the finalized epoch from head may go down but latest cached finalized epoch can never go down